DNN Platform/HttpModules/Membership/MembershipModule.cs

```diff
@@ -256,5 +257,38 @@ public static void AuthenticateRequest(HttpContextBase context, bool allowUnknow
                 context.Items.Add("UserInfo", new UserInfo());
             }
         }
+
+        private static bool NeedLogout(HttpContextBase context, UserInfo user)
+        {
+            try
+            {
+                if (user == null || user.IsDeleted || user.Membership.LockedOut
+                    || !user.Membership.Approved && !user.IsInRole("Unverified Users")
+                    || !user.Username.Equals(context.User.Identity.Name, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return true;
+                }
+
+                // if user's password changed after the user cookie created, then force user to login again.
+                var authCookie = context.Request.Cookies[FormsAuthentication.FormsCookieName];
```

Comment:
I am concerned with regards to the performance of this change.  As a HttpModule this is going to execute once for every managed request, including .axd and other style files.  This introduces a number of string operations, an additional decryption, and a lookup validation.  

I would prefer that we find another method to accomplish this goal.  Additionally, this is a change in behavior that could be hard to diagnose for users.

Reply:
Hi @mitchelsellers , the requests filter was done at begining of AuthenticateRequest method, it won't process for axd or other files.

Reply:
@zyhfish Ok, this still is a per-page-load situation, I really don't want this overhead and processing on every request

Reply:
Update the code to get issue date from identity instead of from cookie.

Reply:
Can you please revisit @mitchelsellers? The only performance degradation I see now is the `try...catch` block.